### PR TITLE
Render urlEntities such that copying and pasting them will result in cop...

### PR DESCRIPTION
**Pull request by jsha**
...ying and pasting the expandedUrl instead of displayUrl, using invisible (font-size:0) text.

### CODE REVIEW

**Comment by keitaf** [comment]
Does this trick work by itself? Or does it require some additional code, like onClick() hook mentioned in the comment?

I tested the sample HTML in the comment but it didn't work for me. (I get "..." and also a white space in the middle when copied & pasted the displayed URL.)

If it doesn't work by itself, I'd prefer to modify twitter-text to support more flexible option so that we can use it to generate a necessary HTML.

**Comment by jsha** [comment]
This trick works by itself, with the exception you noted that a "... " appears in the copied text.  However, this is a big improvement over the current technique.

New technique would copy naturally as:

"hi ... http://longdomainname.com/foo" # full, working link

Old technique would copy naturally as:

"hi ...nname.com/foo" # broken link

Adding an onCopy handler can further improve the new technique so the ellipsis doesn't get copied at all. ("hi http://longdomainname.com/foo").

I did consider adding a callback parameter so the client code could decide exactly how to render the urlEntities.  But I think this is the best way to render urlEntitites by default, so that even without special JS, URLs get copied in full.  Maybe I should document more clearly how to add an appropriate onCopy handler to hide the ellipses?

**Comment by keitaf** [comment]
I see. Yes this will definitely a big improvement.

Can you please make the similar change on twitter-text-rb? I'll try to make the similar change on twitter-text-java.

**Comment by jsha** [comment]
Will do! Thanks.

File: test/tests.js [hunk 1]
```diff
@@ -70,11 +70,23 @@ test("twttr.txt.autolink", function() {
...
     }]
-  }).match(/<a href="http:\/\/t.co\/0JG5Mcq"[^>]+>blog.twitter.com\/2011\/05\/twitte…<\/a>/), 'Use display url from url entities');
+  });
+  ok(autoLinkResult.match(/<a href="http:\/\/t.co\/0JG5Mcq"[^>]+>/), 'Use t.co URL as link target');
+  ok(autoLinkResult.match(/>blog.twitter.com\/2011\/05\/twitte.*…</), 'Use display url from url entities');
+  ok(autoLinkResult.match(/r-for-mac-update.html</), 'Include the tail of expanded_url');
+  ok(autoLinkResult.match(/>http:\/\//), 'Include the head of expanded_url');
+
+  // Insert the HTML into the document and verify that, if copied and pasted, it would get the expanded_url.
+  var div = document.createElement('div');
```
Comment (keitaf) [inline test/tests.js:80]: Can we make this div invisible or remove it after the test is passed??

File: twitter-text.js [hunk 2]
```diff
@@ -451,8 +452,84 @@ if (!window.twttr) {
...
+            //   …
+            // </span>
+            v['invisible'] = "style='font-size:0; line-height:0'";
+            linkText = stringSupplant("<span class='tco-ellipsis'>#{precedingEllipsis}<span #{invisible}>&nbsp;</span></span><span #{invisible}>#{beforeDisplayUrl}</span><span class='js-display-url'>#{displayUrlSansEllipses}</span><span #{invisible}>#{afterDisplayUrl}</span><span class='tco-ellipsis'><span #{invisible}>&nbsp;</span>#{followingEllipsis}</span>", v);
+          }
+        }
+
         for (var k in options) {
-          htmlAttrs += stringSupplant(" #{k}=\"#{v}\" ", {k: k, v: options[k].toString().replace(/"/, "&quot;").replace(/</, "&lt;").replace(/>/, "&gt;")});
+          var val = options[k].toString().replace(/"/, "&quot;").replace(/</, "&lt;").replace(/>/, "&gt;");
```
Comment (keitaf) [inline twitter-text.js]: You can use twttr.txt.htmlEscape() instead.

**Comment by keitaf** [comment]
Looks good to me. Merging to master.

**Pull request merged**